### DISCUSSION TITLE
Error in trigger when accessing properties with long names

### DISCUSSION THREAD

**Discussion by cmglezpdev** [supabase/supabase#26023] [Questions] [answered]
Hello.

When I am in a trigger and try to access a property of the `NEW` object, if this property has a large name, postgres throws an exception telling me that NEW does not have that property. However, if the property has a short name, this is fine.

Example:

For a property with name `short_description`, postgres throws me the following exception:

`Error running SQL: record "new" has no field "short_description"`

This error is produced just with accessing to the property. For example `NEW.short_description`. For names like `type`, `price`, and even `full_description` this doesn't have problems.

**Comment by GaryAustin1** [comment]
Can you show your table definition?  No capital letter, no extra space at end?  Does it occur on another column also?
Never heard of a limit on column names in new.

**Reply by GaryAustin1** [reply]
I can confirm not likely a length issue...
![image](https://github.com/supabase/supabase/assets/54564956/4cc78fac-5baf-4a19-8d5f-ca57f3485737)
```sql
begin
raise log 'long, short = %,%',new.my_very_very_very_very_very_long_name_column, new.id;
return new;
end;
```

**Reply by cmglezpdev** [reply]
Sure.

This is my table.
![image](https://github.com/supabase/supabase/assets/93418478/66479464-7c6c-4914-85de-7af2ec4a66d1)

and the trigger
![image](https://github.com/supabase/supabase/assets/93418478/477e548f-92b7-4fef-a020-0936a8d8cf7c)
![image](https://github.com/supabase/supabase/assets/93418478/ef8b7133-4fe3-4c68-9048-3d6b829c005a)

So, when i run a simple update:
![image](https://github.com/supabase/supabase/assets/93418478/4b88da82-237b-4342-94a4-7e7df239d364)

<hr/>

Now, if i print the `id`, `type`, `formula` in logs:
![image](https://github.com/supabase/supabase/assets/93418478/6cadadd4-e9bc-4430-a445-31c5ea06a06b)
![image](https://github.com/supabase/supabase/assets/93418478/06be1179-4194-456d-8c45-fced76550165)

However, with `shop_description` and `full_description` this works fine.
I don't know.

**Reply by GaryAustin1** [reply, marked as answer]
At the moment I suspecting a space...
![image](https://github.com/supabase/supabase/assets/54564956/56f24d22-24a3-4d2e-8c74-07e7928f3d11)
![image](https://github.com/supabase/supabase/assets/54564956/0e7fa311-c573-47f3-b6fd-8d534665a3ba)
![image](https://github.com/supabase/supabase/assets/54564956/e0004af1-12c6-4dff-9c65-3cb5f8eb467b)
That column has an extra space at end.
You could look in the UI and highlight the name.
Or  you could try `new."price_with_discount ";`  notice the extra space.

**Reply by cmglezpdev** [reply]
Indeed. The error was that, it had a whitespace at the end. 

Thank you very much!!.

**Reply by GaryAustin1** [reply]
Hey, could you say how you created the column initially?  I thought the dashboard trimmed it now, but could be wrong.

**Reply by cmglezpdev** [reply]
Yes. I created the table using the dashboard, and apparently it did not remove the spaces. I don't know if it has anything to do with it, but I'm using cli version `1.106.1`.

**Reply by GaryAustin1** [reply]
![image](https://github.com/supabase/supabase/assets/54564956/2e0be3d6-83df-48b9-8d4b-1e9408850caa)
Current release...
It could have been fixed as your release is over 7 months ago.
